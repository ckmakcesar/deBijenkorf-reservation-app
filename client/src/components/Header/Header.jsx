import React from 'react';
import PropTypes from 'prop-types';

const Header = ({
  text,
}) => {
  // TODO

  return (
    <div className='root'>
      <h1>{text}</h1>

    </div>
  );
};

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
Header.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Header;
