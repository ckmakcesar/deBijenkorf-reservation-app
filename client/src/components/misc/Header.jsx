import React from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';

import styles from '../../styles/misc/Header.module.css';

const Header = ({
  text,
  smaller,
  id,
}) => (
  <div className={styles.headerContainer}>
    <div
      {...(id ? { id: `Header-${id}` } : null)} // optional id - should be unique
      className={clsx(styles.headerText, smaller && styles.reducedHeaderText)}
    >
      {text}
    </div>
  </div>
);

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
Header.propTypes = {
  text: PropTypes.string.isRequired,
  smaller: PropTypes.bool,
  id: PropTypes.string,
};

Header.defaultProps = {
  smaller: false,
  id: '',
}

export default Header;
