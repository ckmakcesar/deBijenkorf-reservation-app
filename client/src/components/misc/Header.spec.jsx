import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Header from './Header';

describe('component:Header', () => {
  it('Should render Header with every optional props absent', () => {
    const wrapper = shallow(
      <Header
        text='This is the very top of the page'
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render Header with every optional props present', () => {
    const wrapper = shallow(
      <Header
        text='This is the very top of the page'
        smaller={true}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
