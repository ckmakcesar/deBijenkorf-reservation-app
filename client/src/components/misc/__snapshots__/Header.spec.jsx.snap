// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:Header Should render Header with every optional props absent 1`] = `
<div
  className="headerContainer"
>
  <div
    className="headerText"
  >
    This is the very top of the page
  </div>
</div>
`;

exports[`component:Header Should render Header with every optional props present 1`] = `
<div
  className="headerContainer"
>
  <div
    className="headerText reducedHeaderText"
  >
    This is the very top of the page
  </div>
</div>
`;
